bpf/process/generic_calls.h

```diff
@@ -89,6 +101,25 @@ generic_process_event(void *ctx, struct bpf_map_def *heap_map,
 	a = (&e->a0)[index];
 	total = e->common.size;
 
+	if (index <= 4) {
+		btf_config = (&config->btf_arg_child0)[index];
+		struct extract_arg_data extract_data = {
+			.config = config,
+			.btf_config = btf_config,
+			.arg = &a,
+		};
+#ifndef __V61_BPF_PROG
+#pragma unroll
+		for (int i = 0; i < MAX_BTF_ARG_CHILD_DEPTH; ++i) {
+			if (extract_arg_depth(i, &extract_data))
+				break;
+		}
+#else
+		loop(MAX_BTF_ARG_CHILD_DEPTH, extract_arg_depth, &extract_data, 0);
+#endif /* __V61_BPF_PROG */
+		probe_read((void *)&a, sizeof(char *), (char *)a);
```

Comment:
what's the purpose of this?

Reply:
This is needed because pointer are dereferenced on the next iteration when called, not when we do math ptr.
To give more context, let's assume we have linux_binprm and want to follow this `linux_binprm.file.my_var` with `my_var` as string and `file` as `struct file *`. The `is_pointer` check in the loop is not done at `file` iteration but on the next with `my_var`. In order to get this pointer delay back, we make sure `my_var` is safe ptr.

In the tetragon logs it look like this

With this final `probe_read`
```json
{
  "process_lsm": {
    "args": [
      {
        "string_arg": "ls"
      }
    ],
  },
}
```

Without it
```json
    {
  "process_lsm": {
    "args": [
      {
        "string_arg": "xB�O�=�O�ls"
      }
    ],
  },
}
```